### PR TITLE
Documentation Tweaks

### PR DESCRIPTION
**Pull request by JamesPeiris**
Hey everyone! 👋  Just wanted to preface this with the fact I'm getting so much value out of this - really fantastic resource!

This PR contains some minor amendments I feel would have improved my understanding as I followed the tutorials.
Things like removing an extra char in an example, rewording some of the lengthier sentences and keeping tenses consistent to improve how some sentences read.

## Changes
- Fixed [rogue parenthesis](https://github.com/quii/learn-go-with-tests/pull/461/files#diff-46dbb8fc8f2f62f02d43f7be721f173eb9ee2308dd08b8c9e3162109f06062e3L220) in refactored map example
- Split wordier [sentences](https://github.com/quii/learn-go-with-tests/pull/461/files#diff-a70fdf2a914c72f660fff3de45b83704ce07a58533f4976c4fc0e0939697b7daR42-R43) apart
- Reworded some examples ([[1]](https://github.com/quii/learn-go-with-tests/pull/461/files#diff-f91da33bcb77a7302f99e03540deaa92442cf329c561ac4ab374eadb1016ef99R142), [[2]](https://github.com/quii/learn-go-with-tests/pull/461/files#diff-f9d21f97b3794f7e9f3ecbee18666c308cd16b09e6eff91d2c36b6fec6152bf3R179-R181), [[3]](https://github.com/quii/learn-go-with-tests/pull/461/files#diff-f91da33bcb77a7302f99e03540deaa92442cf329c561ac4ab374eadb1016ef99R153-R154)) and [sentences](https://github.com/quii/learn-go-with-tests/pull/461/files#diff-a70fdf2a914c72f660fff3de45b83704ce07a58533f4976c4fc0e0939697b7daR97) for extra clarity
- Improved [tense](https://github.com/quii/learn-go-with-tests/pull/461/files#diff-a70fdf2a914c72f660fff3de45b83704ce07a58533f4976c4fc0e0939697b7daR430) consistency
- Added some missing [punctuation](https://github.com/quii/learn-go-with-tests/pull/461/files#diff-62471b2d0a0688a5529e1c7178be88a5cfdf97c842d2d28c504dbbac66ee5235R547)
- Added missing [backticks](https://github.com/quii/learn-go-with-tests/pull/461/files#diff-a70fdf2a914c72f660fff3de45b83704ce07a58533f4976c4fc0e0939697b7daR278) around code references
- Moved [text](https://github.com/quii/learn-go-with-tests/pull/461/files#diff-019ca03e9ec8796ce4c7e9a01c115a89f55afe444303e5ef665292774a2f99dcL499) to more appropriate location

### CODE REVIEW

File: arrays-and-slices.md [hunk 1]
```diff
@@ -224,12 +223,10 @@ In our case, you can see that having two tests for this function is redundant.
 If it works for a slice of one size it's very likely it'll work for a slice of
 any size \(within reason\).
 
-Go's built-in testing toolkit features a [coverage
-tool](https://blog.golang.org/cover), which can help identify areas of your code
-you have not covered. I do want to stress that having 100% coverage should not
-be your goal, it's just a tool to give you an idea of your coverage. If you have
```
Comment (JamesPeiris) [inline arrays-and-slices.md:230]: `it's just a tool to give you an idea of your coverage`
is essentially a repeat of
`a coverage tool, which can help identify areas of your code you have not covered`
so have reworded to collapse this block without losing any meaning.

File: http-server.md [hunk 2]
```diff
@@ -491,12 +491,12 @@ func main() {
 ```
 
 If you run `go build` again and hit the same URL you should get `"123"`. Not great, but until we store data that's the best we can do.
+It also didn't feel great that our main application was starting up but not actually working. We had to manually test to see the problem.
 
 We have a few options as to what to do next
 
 -   Handle the scenario where the player doesn't exist
 -   Handle the `POST /players/{name}` scenario
--   It didn't feel great that our main application was starting up but not actually working. We had to manually test to see the problem.
```
Comment (JamesPeiris) [inline http-server.md:499]: This isn't really an option, more a description of a current drawback. Moved a few lines above

File: maps.md [hunk 3]
```diff
@@ -217,7 +217,6 @@ t.Run("unknown word", func(t *testing.T) {
 
     assertError(t, got, ErrNotFound)
 })
-}
```
Comment (JamesPeiris) [inline maps.md:220]: Rogue bracket in example

**Comment by quii** [comment]
Hey @JamesPeiris 

Sorry for the delay, just got to reading your changes with @ruthmoog and they look great. Thanks!

**Pull request merged**